packages/cli/src/modules/mcp/mcp-oauth-authorization-code.service.ts

```diff
@@ -81,16 +81,18 @@ export class McpOAuthAuthorizationCodeService {
 	): Promise<AuthorizationCode> {
 		const authRecord = await this.findAndValidateAuthorizationCode(authorizationCode, clientId);
 
-		if (authRecord.used) {
-			await this.authorizationCodeRepository.remove(authRecord);
-			throw new Error('Authorization code already used');
-		}
-
 		if (redirectUri && authRecord.redirectUri !== redirectUri) {
 			throw new Error('Redirect URI mismatch');
 		}
 
-		await this.authorizationCodeRepository.update({ code: authorizationCode }, { used: true });
+		const result = await this.authorizationCodeRepository.update(
+			{ code: authorizationCode, used: false },
+			{ used: true },
+		);
+
+		if (result.affected === 0) {
```

Comment:
For consistency, could we use the same pattern here as in the other one? I.e.

```
			const numAffected = result.affected ?? 0;
			if (numAffected < 1) {
```

Typeorm should return num affected rows for both PG and SQLite, but just to be certain